scripts/genSolanaAPI.js

```diff
@@ -0,0 +1,192 @@
+/* eslint-disable no-console */
+const fs = require('fs');
+const path = require('path');
+const fetchEndpoints = require('./utils/apiUtils');
+
+// URL will be changed when api is deployed
+const API_HOST = 'solana-gateway.moralis.io';
+const SWAGGER_PATH = '/api-json';
+
+const OUTPUT_DIRECTORY = '../src';
+const OUTPUT_FILENAME = 'MoralisSolanaApi.js';
+
+// const BodyParamTypes = {
+//   setBody: 'set body',
+//   property: 'property',
+// };
+
+let content = '';
+
+content += `/**
+ * Automatically generated code, via genSolanaAPI.js
+ * Do not modify manually
+ */
+`;
+
+content += `const axios = require('axios');\n`;
+
+content += `
+class SolanaApi {
+  // URL will be changed when api is deployed
+  static baseURL = 'https://solana-gateway.moralis.io';
+  static BodyParamTypes = {
+    setBody: 'set body',
+    property: 'property',
+  };
+  static initialize({apiKey, serverUrl, Moralis = null}) {
+    if (!serverUrl && !apiKey) {
+      throw new Error('SolanaApi.initialize failed: initialize with apiKey or serverUrl');
+    }
+    if(apiKey) this.apiKey = apiKey;
+    if(serverUrl) this.serverUrl = serverUrl;
+    this.Moralis = Moralis;
+  }
+
+    static getBody(params, bodyParams) {
+  if (!params || !bodyParams || !bodyParams.length) {
+    return undefined;
+  }
+  let body = {};
+  bodyParams.forEach(({ key, type, required }) => {
+    if (params[key] === undefined) {
+      if (required) throw new Error(\`param \${key} is required!\`);
+    } else if (type === this.BodyParamTypes.setBody) {
+      body = params[key];
+    } else {
+      body[key] = params[key];
+    }
+    // remove the param so it doesn't also get added as a query param
+    delete params[key];
+  });
+  return body;
+}
+
+static getParameterizedUrl(url, params) {
+  if (!Object.keys(params).length) return url;
+
+  // find url params, they start with :
+  const requiredParams = url.split('/').filter(s => s && s.includes(':'));
+  if (!requiredParams.length) return url;
+
+  let parameterizedUrl = url;
+  requiredParams.forEach(p => {
+    // strip the : and replace with param value
+    const key = p.substr(1);
+    const value = params[key];
+    if (!value) {
+      throw new Error(\`required param \${key} not provided\`);
+    }
+    parameterizedUrl = parameterizedUrl.replace(p, value);
+
+    // remove required param from param list
+    // so it doesn't become part of the query params
+    delete params[key];
+  });
+
+  return parameterizedUrl;
+}
+
+static getErrorMessage(error, url) {
+  return (
+    error?.response?.data?.message ||
+    error?.message ||
+    error?.toString() ||
+   \`Solana API error while calling \${url}\`
+  );
+}
+
+static async fetch({ endpoint, params }) {
+  const { method = 'GET', url, bodyParams } = endpoint;
+  if(this.Moralis) {
+      const { web3 } = this.Moralis;
+      
+      if (!params.address && web3) {
+        params.address = await (await web3.eth.getAccounts())[0];
+      }
+    }
```

Comment:
```suggestion
      if (!params.address) {
        params.address = this.Moralis.account;
      }
    }
```

This logic has changed, as we don't have web3.eth anymore